src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/conditions/ElectronicMonitoringProgrammeService.kt

```diff
@@ -0,0 +1,119 @@
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.Licence
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.UpdateElectronicMonitoringProgrammeRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.ElectronicMonitoringProviderRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.StaffRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.AuditService
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.policies.LicencePolicyService
+
+@Service
+class ElectronicMonitoringProgrammeService(
+  private val licenceRepository: LicenceRepository,
+  private val licencePolicyService: LicencePolicyService,
+  private val auditService: AuditService,
+  private val staffRepository: StaffRepository,
+  private val electronicMonitoringProviderRepository: ElectronicMonitoringProviderRepository,
+) {
+
+  @Transactional
+  fun updateElectronicMonitoringProgramme(licenceId: Long, request: UpdateElectronicMonitoringProgrammeRequest) {
+    val licenceEntity = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val username = SecurityContextHolder.getContext().authentication.name
+    val staffMember = staffRepository.findByUsernameIgnoreCase(username)
+
+    val electronicMonitoringProvider = when (licenceEntity) {
+      is CrdLicence -> requireNotNull(licenceEntity.electronicMonitoringProvider) {
+        "ElectronicMonitoringProvider is null for CrdLicence: $licenceId"
+      }
+
+      is HdcLicence -> requireNotNull(licenceEntity.electronicMonitoringProvider) {
+        "ElectronicMonitoringProvider is null for HdcLicence: $licenceId"
+      }
+
+      else -> error("Trying to update electronic monitoring provider details for non-crd or non-hdc: $licenceId")
+    }
+    electronicMonitoringProvider.isToBeTaggedForProgramme = request.isToBeTaggedForProgramme
+    electronicMonitoringProvider.programmeName = request.programmeName
+    licenceRepository.saveAndFlush(licenceEntity)
+
+    auditService.recordAuditEventUpdateElectronicMonitoringProgramme(licenceEntity, request, staffMember)
+  }
+
+  @Transactional
+  fun handleElectronicMonitoringResponseRecords(licenceEntity: Licence) {
+    val additionalApConditions = licenceEntity.additionalConditions.filter { it.conditionType == "AP" }
+    val conditionsRequiringElectronicMonitoringResponse = licencePolicyService.getConditionsRequiringElectronicMonitoringResponse(licenceEntity.version!!, additionalApConditions.map { it.conditionCode })
+    if (conditionsRequiringElectronicMonitoringResponse.isNotEmpty()) {
+      LicenceConditionService.log.info("Handling Electronic Monitoring response record for conditions: ${conditionsRequiringElectronicMonitoringResponse.joinToString(",") { it.code }}")
```

Comment:
don't want to be referencing the log from condition service, there should be one here 
(ideally we'd keep them private)